packages/ember-htmlbars/lib/component.js

```diff
@@ -465,7 +471,8 @@ const Component = View.extend(TargetActionSupport, ActionSupport, {
 Component[NAME_KEY] = 'Ember.Component';
 
 Component.reopenClass({
-  isComponentFactory: true
+  isComponentFactory: true,
+  positionalParams: []
```

Comment:
so I mean to move the doc here too, but the issue is `if (positionalParams) {` will optimize to false because this is uncommon but not exceptional and the branch deopts, I was doing a lot of clean as I go, I can remove that stuff and make it a separate PR